beep/structure/base.py

```diff
@@ -1568,3 +1574,19 @@ def get_CV_current(CV):
     if isinstance(CV, pd.DataFrame):
         if len(CV):
             return(CV.current.iat[-1])
+
+
+def get_CV_capacity(CV):
+    """
+    Helper function to compute CV capacity.
+
+    Args:
+        CV (pd.DataFrame): CV segement of charge
+
+    Returns:
+        (float): charge capacity during the CV segment
+
+    """
+    if isinstance(CV, pd.DataFrame):
+        if len(CV):
```

Comment:
same with this length check, I am not sure we'd want to return `None` here if the df is empty?

Reply:
These checks are handling rare instances where get_CV_segment_from_charge() fails to extract a CV segment (maybe it doesn't exist, or tolerances need to be tweaked) and returns None or and empty DataFrame. When this happens, the statistic gets a value of None (which is helpful for debugging) and the data structuring just moves on to the next cycle.

Can you think of a better way to passively handle this type of failure?

Reply:
I think the question to be answered is whether the failure should be handled passively or actively. 

If it is passive, I would advocate for a minimum of an ERROR level log message.

In my experience, the most difficult types of failures to debug are those that fail silently, then require backtracking through the code to see where a certain value gets assigned. I'd generally vote for an exception, warning, or log message to be thrown

Reply:
@d-cogswell Let me know what you decide on and make any consequent changes and I'll merge in! 

Note if you decide to use a log ERROR for passive failures, you can easily import the beep one with 

`from beep import logger`

Then in an `else` block below this `if`, we'd have something like

`logger.error("No CV segment exists for <<parameters useful for debugging?>>!")`

Reply:
I can try moving the error checking to the CV extraction function. Should it be a warning rather than an error? Don't errors throw an exception be default?

Reply:
I don't believe so if it is within the logging framework, but I'll leave that choice (warn vs. error) up to you!

Reply:
@d-cogswell This looks ok to me now, let me know if you are ready for me to merge it

Reply:
Thanks @ardunn , it's ready to merge.